### PR TITLE
Added Queue

### PR DESCRIPTION
**Pull request by abhstract**
#### References to other Issues or PRs or Relevant literature

Fixes https://github.com/codezonediitj/pydatastructs/issues/22

### CODE REVIEW

File: pydatastructs/Queue.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from __future__ import print_function
+from __
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: ?

File: pydatastructs/Queue.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+
+
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: Keep only one new line.

File: pydatastructs/Queue.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+class Queue():
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: No `()`

File: pydatastructs/Queue.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	No particular raisse till now.
+
+	Operations available:
+	=====================
+	1. initialisation of queue.
+	2. Check whether Queue is empty or not.
+	3. Appending elements into the queue on a first come first serve basis.
+	4. popping elements on first arrival basis.
+	5. detecting length of Queue.
+	6. Check whether queue is full or not.
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: Not needed.

**Comment by czgdp1807** [comment]
Please see the coding style in other modules and make suitable changes. Put the `queue.py` file in https://github.com/codezonediitj/pydatastructs/tree/master/pydatastructs/miscellaneous_data_structures

File: pydatastructs/Queue.py (hunk 4 above)
Comment (abhstract) [reply pydatastructs/Queue.py]: thanks

Comment (abhstract) [reply pydatastructs/Queue.py]: Will I have to write anything in the __init__.py file?

File: pydatastructs/Queue.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	 '''
+	
+    def __init__(self):
+        self.elements = []
+
+	def len(self):
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: `def __len__(self):`

File: pydatastructs/Queue.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	=====================
+	1. initialisation of queue.
+	2. Check whether Queue is empty or not.
+	3. Appending elements into the queue on a first come first serve basis.
+	4. popping elements on first arrival basis.
+	5. detecting length of Queue.
+	6. Check whether queue is full or not.
+	 '''
+	
+    def __init__(self):
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: `def __new__(self):`

File: pydatastructs/Queue.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        self.elements = []
+
+	def len(self):
+        return len(self.elements)
+    
+	def isEmpty(self):
+        return self.elements == []
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: Remove this method.

File: pydatastructs/Queue.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+    def isFull(self):
+    	if len(self.elements)<self.max_size 
+    		return 0 
+    	else return 1
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: Remove this method.

File: pydatastructs/Queue.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+    def append(self, elem):
+    	if (isFull(self)!=0)
+    		self.elements.insert(0,elem)
+    	else raise Error("Unable to acess memory segment")
+
+	def popleft(self):
+        if isEmpty(self)
+        	 return "Queue empty"
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: use `raise ValueError("Queue empty")`.

File: pydatastructs/Queue.py (hunk 4 above)
Comment (czgdp1807) [reply pydatastructs/Queue.py]: Yes, see how stack is added to `__init__.py` file. Follow a similar pattern.

File: pydatastructs/Queue.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+	 '''
+	
+    def __new__(self):
+        self.elements = []
+
+	def __len__(self):
+        return len(self.elements)
+    
+    def append(self, elem):
+   		self.elements.insert(0,elem)
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: Indentation issue.

File: pydatastructs/Queue.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  
+
+	def popleft(self):
+		self.elements.pop()
```
Comment (czgdp1807) [inline pydatastructs/Queue.py]: Indentation issue.

**Comment by czgdp1807** [comment]
In reference to the PR policy, this PR will be closed tomorrow. Thanks for the contributions. If in future you want to resume your work on it then please comment below. We will re-open it.

**Comment by utkarsh0702** [comment]
Can I add the queue.py file

**Pull request closed without merging**